crates/driver/src/tests/cases/protocol_fees.rs

```diff
@@ -75,9 +77,16 @@ async fn protocol_fee_test_case(test_case: TestCase) {
             .then_some(test_case.execution.solver.sell - solver_fee),
     };
     // Amounts expected to be returned by the driver after fee processing
-    let expected_amounts = ExpectedOrderAmounts {
-        sell: test_case.execution.driver.sell,
-        buy: test_case.execution.driver.buy,
+    let expected_amounts = if test_case.fee_handler == FeeHandler::Driver {
+        ExpectedOrderAmounts {
+            sell: test_case.execution.driver.sell,
+            buy: test_case.execution.driver.buy,
+        }
+    } else {
+        ExpectedOrderAmounts {
+            sell: test_case.execution.solver.sell,
+            buy: test_case.execution.solver.buy,
+        }
```

Comment:
I am not sure we need this. AFAIU we always want to assert the value from the driver. Can @squadgazzz maybe comment on this as he knows better the test infra.

Reply:
> I am not sure we need this. AFAIU we always want to assert the value from the driver. Can @squadgazzz maybe comment on this as he knows better the test infra. (might be related to comment 👇 )

Reply:
My idea was to directly use the `solver` solution since it has to match the expected order amounts for `FeeHanlder::Solver`, in case in the future someone makes a tests with it and make a mistake. But as you suggested on the comment below, you are right, it is better to have it always from the driver and adjusts the amounts in the test itself.